camel/agents/chat_agent.py

```diff
@@ -281,97 +281,73 @@ def step(
         """
         messages = self.update_messages('user', input_message)
 
-        output_messages: Optional[List[BaseMessage]]
+        loop_msg: bool = True
```

Comment:
@Obs01ete I think this is necessary due to the fact that function calling response and normal text response are different. Given our input, if there is a function needed to be call, the model will return a response with function arguments, function name, and **a `finish_reason` being "function_call" instead of the normal "stop"**. We then execute the function and pack the information of function name, arguments and execution results into messages. These messages are sent back to the model again, and model will repeat the process again but taking the last function call into consideration. The `step` should only be terminated when the model thinks there is no need to call any more function and return a response with the `finish_reason` being "stop". The last response will be a normal text message summarizing the previous function calls.